nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/factory/Nd4j.java

```diff
@@ -406,9 +406,34 @@ public static INDArray expandDims(INDArray input, int dimension) {
         long[] indexes = new long[input.rank() + 1];
         for (int i = 0; i < indexes.length; i++)
             indexes[i] = i < dimension ? shape[i] : i == dimension ? 1 : shape[i - 1];
-        return input.reshape(input.ordering(), indexes);
+        return input.reshape(indexes);
     }
 
+    /**
+     * Squeeze : removes a dimension of size 1
+     * @param input the input array
+     * @param dimension the dimension to remove
+     * @return the array with dimension removed
+     */
+    public static INDArray squeeze(INDArray input, int dimension) {
+        if (dimension < 0){
+            dimension += input.rank();
+        }
+        long[] shape = input.shape();
+        if (shape[dimension] != 1){
+            throw new ND4JIllegalStateException("Can squeeze only dimensions of size 1.");
```

Comment:
Couple of things here:
(a) we can use Preconditions.checkArgument/checkState - less verbose
(b) When throwing exceptions, I think it's good practice to include useful information. What dimension? What's the shape of the array? Without that, I need a debugger to get that information, which adds a lot of time required to fix it...